bugrap/src/main/java/org/vaadin/bugrap/BaseModel.java

```diff
@@ -4,19 +4,22 @@
 import org.vaadin.bugrap.domain.entities.Reporter;
 
 import com.vaadin.navigator.Navigator;
+import com.vaadin.server.VaadinSession;
 
 public class BaseModel {
 
 	public static final String NAV_LOGIN = "login";
 	public static final String NAV_REPORT = "report";
-	// public static final String NAV_REPORTDET = "detail";
+	
+	private static final String SESSIONVAR_LOGINUSER = "loginuser";
 
 	private static final BugrapRepository repository = new BugrapRepository(DatabaseHelper.DB_LOCATION);
 	private Navigator navigator;
-	public static Reporter loginUser;
+	private VaadinSession session;
 
-	public BaseModel(Navigator nav) {
+	public BaseModel(Navigator nav, VaadinSession vs) {
```

Comment:
There's a way of getting the current Vaadin session, so you shouldn't need to be passing it around.